packages/neuron-ui/src/components/History/index.tsx

```diff
@@ -46,51 +73,343 @@ const History = () => {
+      dataIndex: 'amount',
+      align: 'left',
+      isBalance: true,
+      minWidth: '150px',
+      render(_, __, item, show) {
+        let amount = '--'
+        let sudtAmount = ''
+        let isReceive = false
+
+        if (item.blockNumber !== undefined) {
+          if (item.nftInfo) {
+            // NFT
+            const { type, data } = item.nftInfo
+            amount = show ? `${type === 'receive' ? '+' : '-'}${nftFormatter(data)}` : `${HIDE_BALANCE}mNFT`
+            isReceive = type === 'receive'
+          } else if (item.sudtInfo?.sUDT) {
+            if (item.sudtInfo.sUDT.decimal) {
+              sudtAmount = sUDTAmountFormatter(sudtValueToAmount(item.sudtInfo.amount, item.sudtInfo.sUDT.decimal))
+            }
+          } else {
+            amount = show ? `${shannonToCKBFormatter(item.value, true)} CKB` : `${HIDE_BALANCE} CKB`
+            isReceive = !amount.includes('-')
+          }
+        }
+        return sudtAmount ? (
+          <>
+            {show ? sudtAmount : HIDE_BALANCE}&nbsp;
+            <UANTonkenSymbol
+              className={styles.symbol}
+              name={item.sudtInfo!.sUDT.tokenName}
+              symbol={item.sudtInfo!.sUDT.symbol}
+            />
+          </>
+        ) : (
+          <span className={show && isReceive ? styles.isReceive : ''}>{amount}</span>
+        )
+      },
+    },
+    {
+      title: t('history.table.timestamp'),
+      dataIndex: 'timestamp',
+      align: 'left',
+      render: (_, __, item) => uniformTimeFormatter(item.timestamp),
+    },
+    {
+      title: t('history.table.status'),
+      dataIndex: 'status',
+      align: 'left',
+      render(_, __, item) {
+        const confirmations = 1 + bestBlockNumber - +item.blockNumber
+        let status = item.status as string
+        if (status === 'success' && confirmations < CONFIRMATION_THRESHOLD) {
+          status = 'confirming'
+        }
+
+        return t(`history.${status}`)
+      },
+    },
+    {
+      title: t('history.table.operation'),
+      dataIndex: 'operation',
+      align: 'left',
+      render(_, idx, ___, ____, expandedRow) {
+        return <ArrowOpenRightIcon className={styles.arrow} data-is-open={expandedRow === idx} />
+      },
+    },
+  ]
+
+  const onActionBtnClick = useCallback(
+    (e: React.SyntheticEvent<HTMLButtonElement>) => {
+      const btn = (e.target as HTMLButtonElement)?.closest('button')
+      if (btn?.dataset?.hash && btn?.dataset?.action) {
+        switch (btn.dataset.action) {
+          case 'explorer': {
+            openExternal(`${getExplorerUrl(isMainnet)}/transaction/${btn.dataset.hash}`)
+            break
+          }
+          case 'detail': {
+            navigate(`${RoutePath.HistoryDetailPage}/${btn.dataset.hash}`)
+            break
+          }
+          default: {
+            // ignore
+          }
+        }
+      }
+    },
+    [isMainnet]
+  )
+
+  const rowExtendRender = (column: State.Transaction) => {
```

Comment:
Can we extract the `rowExtendRender` as a function component out of the current component?

Reply:
Maybe to be a function it will be more convenient to pass the data from the `Table`.